### PR TITLE
Fixes for episodeDetails

### PR DESCRIPTION
**Pull request by mre**
Not all fields get sent by Apple; provide defaults as fallback.
This affects the code and the DB.

### CODE REVIEW

**Comment by mre** [comment]
Corresponding DB deploy request: https://app.planetscale.com/services-openpodcast/openpodcast/deploy-requests/31

File: db_schema/schema.sql [hunk 1]
```diff
@@ -156,11 +156,11 @@ DROP TABLE IF EXISTS appleEpisodeDetails;
 CREATE TABLE appleEpisodeDetails (
   account_id INTEGER NOT NULL,
   episode_id BIGINT NOT NULL,
-  aed_playscount INTEGER NOT NULL,
-  aed_totaltimelistened BIGINT NOT NULL,
-  aed_uniqueengagedlistenerscount INTEGER NOT NULL,
-  aed_uniquelistenerscount INTEGER NOT NULL,
-  aed_engagedplayscount INTEGER NOT NULL,
+  aed_playscount INTEGER,
```
Comment (woolfg) [inline db_schema/schema.sql:159]: why did you remove NOT NULL? do you set it to NULL, it should be just 0, right?

File: src/schema/apple/episodeDetails.json [hunk 2]
```diff
@@ -38,17 +38,7 @@
           "type": "number",
           "default": 0
         }
-      },
```
Comment (woolfg) [inline src/schema/apple/episodeDetails.json:41]: I think we can add this again, if we convince ajv to not only validate the schema but also edit the data:
see https://ajv.js.org/options.html#usedefaults

maybe we can add this in `src/schema/SchemaValidator.ts`

File: db_schema/schema.sql (hunk 1 above)
Comment (woolfg) [reply db_schema/schema.sql:159]: if we convince ajv to not only validate the schema but also edit the data, this should be obsolete (see other comment including the link to the option for ajv)

File: fixtures/appleEpisodeDetailsSparse.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      "playscount": 53.0,
+      "podcastid": "1603082924",
+      "totaltimelistened": 42716.0,
+      "uniqueengagedlistenerscount": 15.0,
+      "uniquelistenerscount": 20.0,
+      "engagedplayscount": 16.0
+    },
+    "episodePlayHistogram": [
+      {
+        "2865": 11.0
```
Comment (woolfg) [inline fixtures/appleEpisodeDetailsSparse.json:56]: that looks really broken, 11 listeners until 2865? but yeah, I guess we have to deal with this.

**Review by woolfg — APPROVED** [review]
lgtm

File: src/schema/apple/episodeDetails.json (hunk 2 above)
Comment (mre) [reply src/schema/apple/episodeDetails.json]: true; added the option and reverted the schema changes. Tested and works

File: db_schema/schema.sql (hunk 1 above)
Comment (mre) [reply db_schema/schema.sql]: true; added the option and reverted the schema changes. Tested and works.

**Comment by mre** [comment]
Added the option to AJV and reverted the schema changes. They are no longer needed because AJV fills in the missing defaults. 👍 Tested and works with empty perf and non-empty perf data.

**Review by woolfg — APPROVED** [review]
looks still good to me, meeerge i
